RFS/src/main/java/org/opensearch/migrations/bulkload/common/OpenSearchClientFactory.java

```diff
@@ -0,0 +1,45 @@
+package org.opensearch.migrations.bulkload.common;
+
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.opensearch.migrations.Version;
+import org.opensearch.migrations.VersionMatchers;
+import org.opensearch.migrations.bulkload.common.http.ConnectionContext;
+import org.opensearch.migrations.bulkload.version_es_6_8.OpenSearchClient_ES_6_8;
+import org.opensearch.migrations.bulkload.version_os_2_11.OpenSearchClient_OS_2_11;
+import org.opensearch.migrations.reindexer.FailedRequestsLogger;
+
+@RequiredArgsConstructor
+@Slf4j
+public class OpenSearchClientFactory {
+    // Version can be null, and if so, the "default" client, for OS_2_11 will be provided (matching the pre-factory behavior)
+    private final Version version;
+
+    public OpenSearchClient get(
+        ConnectionContext connectionContext
+    ) {
+        if (version == null || VersionMatchers.isOS_1_X.test(version) || VersionMatchers.isOS_2_X.test(version) || VersionMatchers.isES_7_X.test(version)) {
```

Comment:
Using `or(...)` with predicates is much more readable IMO
```suggestion
        if (VersionMatchers.isOS_1_X.or(VersionMatchers.isOS_2_X).or(VersionMatchers.isES_7_X).test(version)) {
```

Reply:
I didn't know this syntax! Thanks!